Improve enums docs

rafalp opened the pull request:
Adds more details on enum documentation and mentions that you are supposed to pass `EnumType` instances to `make_executable_schema`.

mat-sop reviewed (commented):
I don't understand, why do you backport changes to 0.4.0 but remove 0.6.0 version altogether?

rafalp commented:
@mat-sop The way versioning works in Docusaurus document from current version is used, and if it's not available, latest version of given document from previous version is used in it's stead. Document from 0.6 was almost the same as one from 0.4, so I've removed 0.6 and updated 0.4, causing every version since 0.4 to use new documentation for enums.
